### PR TITLE
AIP-84: Migrating DELETE queued asset events for assets to fastAPI

### PR DESCRIPTION
**Pull request by vatsrahul1001**
Related: https://github.com/apache/airflow/issues/42370
Dependent: https://github.com/apache/airflow/pull/43934

Swagger Specs
image
image

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/api_fastapi/core_api/routes/public/assets.py [hunk 1]
```diff
@@ -153,3 +175,25 @@ def get_asset(
...
+    responses=create_openapi_http_exception_doc(
+        [
+            status.HTTP_404_NOT_FOUND,
+        ]
+    ),
+)
+def delete_asset_queued_events(
+    uri: str,
+    session: Annotated[Session, Depends(get_session)],
+    before: str = Query(None),
```
Comment (pierrejeambrun) [inline airflow/api_fastapi/core_api/routes/public/assets.py:192]: I think we can use directly `DateTimeQuery` or `DateTimeQuery | None`. (param validation should be done by the route)

File: tests/api_fastapi/core_api/routes/public/test_assets.py [hunk 2]
```diff
@@ -459,3 +467,83 @@ def test_should_respond_404(self, test_client):
...
+        dag_id = dag.dag_id
+        uri = "s3://bucket/key/1"
+        self.create_assets(session=session, num=1)
+        asset_id = 1
+        self._create_asset_dag_run_queues(dag_id, asset_id, session)
+
+        response = test_client.delete(
+            f"/public/assets/queuedEvent/{uri}",
+        )
+
```
Comment (pierrejeambrun) [inline tests/api_fastapi/core_api/routes/public/test_assets.py:538]: Nice we could also check in the db that the record has actually been deleted.

**Review by pierrejeambrun — COMMENTED** [review]
Nice, I'll do a more in depth review once the base branch has been merged.

**Comment by vatsrahul1001** [comment]
closing in favor of https://github.com/apache/airflow/pull/44138

**Pull request closed without merging**